### PR TITLE
Add rule creation UI

### LINKED ISSUE #70: Create a basic UI for rules builder
Basic functionality only, styling can be done later.

### PR DESCRIPTION
**Pull request by thu5**
This PR adds the UI for rule creation. Closes #70.

output

### CODE REVIEW

File: lib/src/rule_creator_component.dart [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+@Component(
+  selector: 'rule-creator',
+  templateUrl: 'rule_creator_component.html',
+  directives: [coreDirectives, formDirectives, BsInput, bsAccordionDirectives],
+  providers: [ClassProvider(RuleService)],
+)
+class RuleCreatorComponent {
+  final RuleService _ruleService;
+  final CredentialService _credentialService;
```
Comment (gg-qinp) [inline lib/src/rule_creator_component.dart:32]: why this is not within `providers` above

**Review by gg-qinp — APPROVED** [review]

File: lib/root_component.html [hunk 2]
```diff
@@ -61,7 +61,16 @@ <h6 *ngIf="showSideloadMessage" class="secondary-msg" debugId="sideload-msg">
...
         <!-- rules builder-->
         <bs-tabx header="{{rulesBuilderTabName}}">
-            <p>Coming soon</p>
+            <div class="jumbotron d-flex align-items-center">
+                <div class="container-fluid">
+                    <!-- rule component-->
+                    <rule-creator></rule-creator>
+                    <!-- sign off button-->
+                    <div class="d-flex justify-content-end">
+                        <small><a debugId="sign-off-btn" (click)="onClick()" href="#">{{signOutButtonName}}</a></small>
```
Comment (cathyang-work) [inline lib/root_component.html]: Why do we use `href="#"` here? Is it going to be replaced by something else later?

**Review by cathyang-work — APPROVED** [review]

File: lib/src/rule_creator_component.dart (hunk 1 above)
Comment (thu5) [reply lib/src/rule_creator_component.dart:32]: Oops, it should be! I'll fix it

File: lib/root_component.html (hunk 2 above)
Comment (thu5) [reply lib/root_component.html]: That code is reused from the query builder, but no it doesn't need to be there. I'll remove it.

**Pull request merged**